src/debugging/coreclr/CommandLineDotNetClient.ts

```diff
@@ -48,6 +60,75 @@ export class CommandLineDotNetClient implements DotNetClient {
             return undefined;
         }
     }
+
+    public async trustAndExportSslCertificate(projectFile: string, hostExportPath: string, containerExportPath: string): Promise<void> {
+        if (CommandLineDotNetClient.KnownConfiguredProjects.has(projectFile)) {
+            return;
+        }
+
+        await this.addUserSecretsIfNecessary(projectFile);
```

Comment:
>addUserSecretsIfNecessary [](start = 19, length = 25)

Shouldn't we prompt the user before futzing with the contents of the csproj file?

Reply:
We don't in VS; additionally SSL will not work without this.

Reply:
In VS we add UserSecretsId during initial scaffolding of the Dockerfile.  Here we're doing it during F5.  And we're going to modify the user's csproj even if they say "no" to the SSL prompt.

Reply:
If they don't want it done that's what the "configureSsl" flag in launch.json is for.

Reply:
I tend to agree with Steven on this. What if you check everything up front, including:
1. Check if UserSecretsId is in csproj
1. Check if password has been set and that the cert matches the password
1. Check if cert is trusted

And only have one generic prompt to essentially "do it all" if you detect anything wrong.

Reply:
Checking the cert and password match would require a crypto library, and I highly doubt it'd be faster than cleaning the slate on the first F5 of the session.

Reply:
@EricJizbaMSFT Here's an alternative that keeps us away from the mess of checking certificate existence / password match / etc.

Instead of doing this on F5, do it via a command. This would make using the VSCode terminal for cert trust viable on Mac. The only thing I'm not sure of is this--can we use a command to manipulate the Docker launch config, in order to add the necessary environment variables? What should we do if no such launch config exists yet and they run the command?

Additionally, it would need to be done per-project, per-machine; i.e. if you sync'd source that was already set up for SSL, you would have to do it again on your machine in order to make it work (And maybe we somehow detect that on F5 and ask? What if they say no on this machine?)

Reply:
[Written just before you commented - not sure I quite understand what you mean by "do it via a command"]

> Checking the cert and password match would require a crypto library, and I highly doubt it'd be faster than cleaning the slate on the first F5 of the session.

Yeah I wouldn't want to use a separate crypto library. I thought there might be a dotnet command that could verify at least part of this for you. Never mind if there isn't a command.

> Other than that, checking if UserSecretsId is in the project and checking trust is exactly what we're doing.

You're never prompting about UserSecretsId, just trusting the cert. @sgreenmsft's original comment said you should prompt before messing with their csproj. I'm agreeing with that and saying perhaps you should combine it with the "cert trust" prompt so that you don't have two separate prompts.

Reply:
As in, one of the Ctrl + Shift + P commands.

Reply:
Doing it as a command, we avoid several problems--
1. Screwing up already-working configuration without permission
2. Wasting time doing things more frequently than actually needed
3. Mac trust nightmare

Reply:
When does the command get run?

Reply:
Whenever the user does it.

Reply:
I think making it a command that users can run at any time is a good idea regardless. But I still don't get how a command alone would be discovered